File: backend/app/schemas/timetable.py
from pydantic import BaseModel
from uuid import UUID


class TimeTableEntry(BaseModel):
    # 타임테이블 단일 항목. LLM팀 크롤링 추출 결과 원본 포맷 그대로 사용
    # (예: {"date": "2023-09-07", "time": null, "stage": "TOUCH", "event": "N.Flying 16:20 - 17:20 (60)"})
    # date/time/stage는 페스티벌 등에서 정보가 없으면 null일 수 있고, 시간 정보가 time이 아니라
    # event 텍스트 안에 포함되는 경우가 많음 (VLM 추출 특성상 time을 항상 분리해내지 못함)
    date: str | None = None
    time: str | None = None
    stage: str | None = None
    event: str


class TimeTableUpdate(BaseModel):
    # 타임테이블 upsert 요청
    contents: list[TimeTableEntry]


class TimeTableResponse(BaseModel):
    model_config = {"from_attributes": True}

    # 타임테이블 조회 응답
    id: UUID
    concert_id: UUID
    contents: list[TimeTableEntry]

File: Prontend/lib/models/timetable.dart
import 'package:flutter/foundation.dart';

// [백엔드 수정]
// time/description(둘 다 non-null) → date/time/stage(전부 nullable)/event(필수)로 스키마 변경.
/// 타임테이블 한 항목. 백엔드 `schemas/timetable.py`의 `TimeTableEntry`와 대응.
@immutable
class TimeTableEntry {
  final String? date;
  final String? time;
  final String? stage;
  final String event;

  const TimeTableEntry({this.date, this.time, this.stage, required this.event});

  factory TimeTableEntry.fromJson(Map<String, dynamic> json) {
    return TimeTableEntry(
      date: json['date'] as String?,
      time: json['time'] as String?,
      stage: json['stage'] as String?,
      event: json['event'] as String,
    );
  }
}

/// `GET /concerts/{concertId}/timetable` 응답. 백엔드 `TimeTableResponse`와 대응.
@immutable
class TimeTableResponse {
  final String id;
  final String concertId;
  final List<TimeTableEntry> contents;

  const TimeTableResponse({
    required this.id,
    required this.concertId,
    required this.contents,
  });

  factory TimeTableResponse.fromJson(Map<String, dynamic> json) {
    return TimeTableResponse(
      id: json['id'] as String,
      concertId: json['concert_id'] as String,
      contents: (json['contents'] as List<dynamic>? ?? const [])
          .map((e) => TimeTableEntry.fromJson(e as Map<String, dynamic>))
          .toList(),
    );
  }
}
